tests/manage/monitoring/conftest.py

```diff
@@ -431,3 +443,423 @@ def corrupt_pg():
+        metadata:
+          name: fio-config
+        data:
+          workload.fio: |
+            # here comes workload configuration
+        """)
+    cm_dict = yaml.safe_load(template)
+    return cm_dict
+
+
+@pytest.fixture
+def fio_job_dict():
+    """
+    Job template for fio workloads.
+    """
+    template = textwrap.dedent("""
+        apiVersion: batch/v1
+        kind: Job
+        metadata:
+          name: fio
+        spec:
+          template:
+            metadata:
+              name: fio
+            spec:
+              containers:
+                - name: fio
+                  image: quay.io/johnstrunk/fs-performance:latest
+                  command:
+                    - "/usr/bin/fio"
+                    - "--output-format=json"
+                    - "/etc/fio/workload.fio"
+                  volumeMounts:
+                    - name: fio-target
+                      mountPath: /mnt/target
+                    - name: fio-config-volume
+                      mountPath: /etc/fio
+              restartPolicy: Never
+              volumes:
+                - name: fio-target
+                  persistentVolumeClaim:
+                    claimName: fio-target
+                - name: fio-config-volume
+                  configMap:
+                    name: fio-config
+        """)
+    job_dict = yaml.safe_load(template)
+    return job_dict
+
+
+def get_storageutilization_size(target_percentage, ceph_pool_name):
+    """
+    For the purpose of the workload storage utilization fixtures, get expected
+    pvc_size based on STORED and MAX AVAIL values (as reported by `ceph df`)
+    for given ceph pool and target utilization percentage.
+
+    This is only approximate, and it won't work eg. if each pool has different
+    configuration of replication.
+
+    Returns:
+        int: pvc_size for storage utilization job (in GiB, rounded)
+    """
+    # get STORED and MAX AVAIL of given ceph pool ...
+    ct_pod = pod.get_ceph_tools_pod()
+    ceph_df_dict = ct_pod.exec_ceph_cmd(ceph_cmd="ceph df")
+    ceph_pool = None
+    ceph_total_stored = 0
+    for pool in ceph_df_dict["pools"]:
+        ceph_total_stored += pool["stats"]["stored"]
+        if pool["name"] == ceph_pool_name:
+            ceph_pool = pool
+    if ceph_pool is None:
+        logger.error((
+            f"pool {ceph_pool_name} was not found "
+            f"in output of `ceph df`: {ceph_df_dict}"))
+    # If the following assert fail, the problem is either:
+    #  - name of the pool has changed (when this happens before GA, it's
+    #    likely ocs-ci bug, after the release it's a product bug),
+    #  - pool is missing (likely a product bug)
+    # either way, the fixture can't continue ...
+    assert ceph_pool is not None, f"pool {ceph_pool_name} should exist"
+    # ... to compute PVC size (values in bytes)
+    total = ceph_pool["stats"]["max_avail"] + ceph_total_stored
+    max_avail_gi = ceph_pool['stats']['max_avail'] / 2**30
```

Comment:
Hmm, I need to check whether OCP is using SI or 2^10*n units for persistent storage.

Reply:
No problem in the code found, the ques as implemented by `get_storageutilization_size` function is less precise because of expected differences between usable space compared to raw space.